java/fury-core/src/main/java/org/apache/fury/serializer/ObjectSerializer.java

```diff
@@ -495,104 +548,165 @@ static boolean writePrimitiveFieldValueFailed(
    * @return true if field value isn't written by this function.
    */
   static boolean writeBasicObjectFieldValueFailed(
-      Fury fury, MemoryBuffer buffer, Object fieldValue, short classId) {
+      Fury fury,
```

Comment:
How about adding a new `writeBasicNullableObjectFieldValueFailed` method? In this method, we handle nullability, and in `writeBasicObjectFieldValueFailed`, all fields are not null